src/sentry/profiles/task.py

```diff
@@ -0,0 +1,125 @@
+from typing import Any, MutableMapping
+
+from django.conf import settings
+
+from sentry.lang.native.symbolicator import Symbolicator
+from sentry.models import Project
+from sentry.profiles.device import classify_device
+from sentry.tasks.base import instrumented_task
+from sentry.utils import json, kafka_config
+from sentry.utils.pubsub import KafkaPublisher
+
+processed_profiles_publisher = None
+
+
+@instrumented_task(  # type: ignore
+    name="profiles.process",
+    queue="profiles.process",
+    default_retry_delay=5,
+    max_retries=5,
+    acks_late=True,
+)
+def process_profile(profile: MutableMapping[str, Any]) -> None:
+    global processed_profiles_publisher
+
+    config = settings.KAFKA_TOPICS[settings.KAFKA_PROFILES]
+    processed_profiles_publisher = KafkaPublisher(
+        kafka_config.get_kafka_producer_cluster_options(config["cluster"]),
+    )
+
+    if profile["platform"] == "cocoa":
+        if not _validate_ios_profile(profile=profile):
+            return None
+        profile = _symbolicate(profile=profile)
+
+    profile = _normalize(profile=profile)
+    processed_profiles_publisher.publish(
+        "processed-profiles",
+        json.dumps(profile),
+    )
+
+
+def _normalize(profile: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
+    normalized_profile = {
+        "device_locale": profile["device_locale"],
+        "device_manufacturer": profile["device_manufacturer"],
+        "device_model": profile["device_model"],
+        "device_os_name": profile["device_os_name"],
+        "device_os_version": profile["device_os_version"],
+        "duration_ns": int(profile["duration_ns"]),
+        "environment": profile.get("environment"),
+        "organization_id": profile["organization_id"],
+        "platform": profile["platform"],
+        "profile": json.dumps(profile["profile"]),
```

Comment:
I transform this field as a string now because it will be stored as a string in ClickHouse. Alternative would be to maybe change the column type in ClickHouse to accept a JSON object or something, so making it a key value store of some sort, but we don’t really need that for profiling.

Reply:
Longer term would this be a scenario where we would want to use BigTable or another Key/Value store to hold the blobs like we do for events?

Reply:
Also curious about this. Not sure how well Clickhouse handles potentially large blobs like this.

Reply:
The plan long term is to not have to store raw information like this but some common frame format into its own table where we can aggregate directly using ClickHouse, something that kinda looks like what's describe here https://www.notion.so/sentry/Improving-the-current-stack-trace-data-model-bc1658f7ecd941f387d51f844f8d4c86 .

Let me know if you want more details about this.